sql/src/test/java/io/crate/testing/SQLTransportExecutor.java

```diff
@@ -365,12 +366,10 @@ private static Object toJdbcCompatObject(Connection connection, Object arg) {
             Collection values = (Collection) arg;
             if (values.isEmpty()) {
                 return null; // TODO: can't insert empty list without knowing the type
-            }
-
-            if (values.iterator().next() instanceof Map) {
+            } else if (values.iterator().next() instanceof Map) {
```

Comment:
Do we even need this custom case handling ?

Wouldn't the regular logic below with the converted values accomplish the same?

Reply:
In this case we will create an array of `json` instead of `object` type, which results in decoding failure. I've to check whether it related to `PGArray` decoding logic or to the way we create the `PgObject` from a json string.